pkg/workflows/workflow.go

```diff
@@ -249,11 +252,19 @@ func (w *workflowLifecycle) submit(ctx context.Context, wp *unstructured.Unstruc
 		if err := controllerutil.SetControllerReference(w.addon, wfv1, w.scheme); err != nil {
 			return addonmgrv1alpha1.Failed, err
 		}
-
-		err = w.Create(ctx, wfv1)
+		wf, err := common.WorkFlowFromUnstructured(wfv1)
 		if err != nil {
+			fmt.Printf("### failed convert unstructure to workflow")
 			return addonmgrv1alpha1.Failed, err
 		}
+		_, err = w.wfclientset.ArgoprojV1alpha1().Workflows(wp.GetNamespace()).Create(ctx, wf, metav1.CreateOptions{})
+		if apierrors.IsAlreadyExists(err) {
+			return addonmgrv1alpha1.Succeeded, nil
+		} else if err != nil {
+			msg := fmt.Sprintf("failed creating wf %s err %v", wp.GetName(), err)
+			fmt.Println(msg)
```

Comment:
Prints are not correct

Reply:
28ea8a9de38232ea59393796dbe97d08d4dd8c6e

Reply:
I still see fmt.Println()

Reply:
f70a7281b87b34b6999fbdcae5db6ccd5f443613